homework-2/p4.java

```diff
@@ -0,0 +1,80 @@
+import java.util.*;
+
+class P4{
+
+	// abababd
+	// ababd
+	static int findLastIndexKMP(String s2, String s1)
```

Comment:
Search for inbuilt methods on String or StringBuilder class to do this operation, instead of manually implementing the logic.